internal/v4/search.go

```diff
@@ -28,7 +25,14 @@ func (h *Handler) serveSearch(_ http.Header, req *http.Request) (interface{}, er
 	if err != nil {
 		return "", err
 	}
-	h.groupsFromRequest(&sp, req)
+	// Ignore the error from checkRequest and carry on assuming no authorization was
+	// sent.
+	auth, _ := h.checkRequest(req)
```

Comment:
Seems like a good approach in general, but I'd
be inclined to log the error (Info level?)
e.g.
log.Infof("authorization failed on search request, granting no privileges: %v", err)

?